### PR TITLE
[MRG+1] Less stringent normalization test for float128.

### PR DESCRIPTION
**Pull request by anntzer**
cf #7159.  I decided to go the less stringent way.

### CODE REVIEW

File: lib/matplotlib/tests/test_colors.py [hunk 1]
```diff
@@ -208,7 +208,9 @@ def test_Normalize():
     # and for scalar ones.
     eps = np.finfo(np.longdouble).resolution
     norm = plt.Normalize(1, 1 + 100 * eps)
-    assert_equal(norm(1 + 50 * eps), .5)
+    # This returns exactly 0.5 when longdouble is extended precision (80-bit),
+    # but only a value close to it when it is quadruple precision (128-bit).
+    assert 0 < norm(1 + 50 * eps) < 1
```
Comment (NelleV) [inline lib/matplotlib/tests/test_colors.py]: You could also use one of the "almost equal" asserts from numpy (or nose, or pytest)

Comment (anntzer) [inline lib/matplotlib/tests/test_colors.py]: It's not almost equal (it's around 0.2% off).

Comment (NelleV) [inline lib/matplotlib/tests/test_colors.py]: I don't understand your comment. The tests "assert_almost_equal" has a epsilon value. It is used exactly for that case where there are floating points error. You need to set the epsilon value yourself.

Comment (anntzer) [inline lib/matplotlib/tests/test_colors.py]: Ah, I was thinking about the default of assert_allclose.  Anyways, see the discussion starting at https://github.com/matplotlib/matplotlib/issues/7159#issuecomment-249054207: the point is that I'd rather let a FP expert confirm what is the tolerance that makes sense; without that the safest is just to check that the normalization is not trivial (does not map everything to 0 or 1).

**Comment by tacaswell** [comment]
backported to v2.x as 2e12ac9

**Pull request merged**